instrumentation/github.com/confluentinc/confluent-kafka-go/kafka/splunkkafka/test/kafka_test.go

```diff
@@ -0,0 +1,325 @@
+// Copyright Splunk Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build cgo && linux
+// +build cgo,linux
+
+package test
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"net"
+	"os"
+	"testing"
+
+	"github.com/confluentinc/confluent-kafka-go/kafka"
+	"github.com/ory/dockertest/v3"
+	"github.com/ory/dockertest/v3/docker"
+	"github.com/signalfx/splunk-otel-go/instrumentation/github.com/confluentinc/confluent-kafka-go/kafka/splunkkafka"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/sdk/trace/tracetest"
+	semconv "go.opentelemetry.io/otel/semconv/v1.4.0"
+	traceapi "go.opentelemetry.io/otel/trace"
+	"go.uber.org/goleak"
+)
+
+var (
+	key, val = []byte("test key"), []byte("test value")
+
+	testGroupID = "gotest"
+	testTopic   = "gotest"
+)
+
+func TestChannelBasedProducer(t *testing.T) {
+	defer goleak.VerifyNone(t)
+
+	partition := int32(0)
+	sr, opts := newFixtures()
+	p := newProducer(t, opts...)
+
+	done := make(chan struct{})
+	var sent *kafka.Message
+	go func() {
+		defer close(done)
+		sent = requireEventIsMessage(t, <-p.Events())
+	}()
+	go func() {
+		p.ProduceChannel() <- &kafka.Message{
+			TopicPartition: kafka.TopicPartition{
+				Topic:     &testTopic,
+				Partition: partition,
+			},
+			Key:   key,
+			Value: val,
+		}
+	}()
+
+	// Wait for the delivery report goroutine to finish.
+	<-done
```

Comment:
sometime a create a "test timeout channel" to prevent the test from deadlocking

we can do it e.g. like this

```go
// ...
  timeout := timeoutCh(t, time.Minute)
// ...
	select {
	case <-timeout:
		t.Fatal("test timeout")
	case <- input:
		// we can put the assertions here or just assign the value to a variable to decrease the indentation
	}
// ...

// timeoutCh prepares a channel that can be used to implement a test timeout.
func timeoutCh(t *testing.T, d time.Duration) <-chan time.Time {
	timeout := time.NewTimer(d)
	t.Cleanup(func() { timeout.Stop() })
	return timeout.C
}
```